controller/dataplane/owned_deployment.go

```diff
@@ -306,17 +355,31 @@ func reconcileDataPlaneDeployment(
 
 		k8sresources.SetDefaultsPodTemplateSpec(&desired.Spec.Template)
 
+		// Keep track of this for logging purposes
+		originalReplicaCount := existing.Spec.Replicas
+
 		// ensure that object metadata is up to date
 		updated, existing.ObjectMeta = k8sutils.EnsureObjectMetaIsUpdated(existing.ObjectMeta, desired.ObjectMeta)
 
-		// some custom comparison rules are needed for some PodTemplateSpec sub-attributes, in particular
-		// resources and affinity.
+		isRestartOperation, restartTimeStr := isRecentDeploymentRestart(&existing.Spec.Template, logger)
+
+		// some custom comparison rules are needed for some PodTemplateSpec sub-attributes
 		opts := []cmp.Option{
 			cmp.Comparer(k8sresources.ResourceRequirementsEqual),
+			utils.IgnoreAnnotationKeysComparer(restartAnnotationKey),
 		}
 
-		// ensure that PodTemplateSpec is up to date
 		if !cmp.Equal(existing.Spec.Template, desired.Spec.Template, opts...) {
+
+			if isRestartOperation {
+				// Preserve the restart annotation
+				if desired.Spec.Template.Annotations == nil {
+					desired.Spec.Template.Annotations = make(map[string]string)
+				}
+				// Use the restart timestamp we already retrieved
+				desired.Spec.Template.Annotations[restartAnnotationKey] = restartTimeStr
+			}
```

Comment:
If we move the log about the found restart annotation in here then we'd be able to call `isRecentDeploymentRestart` in this scope and thus making scope of `isRestartOperation` and `restartTimeStr` that smaller. WDYT?

We'd have to remove logging of `isRestartOperation` from line 411 but I'm OK with that.

Reply:
makes sense